docs/decisions/0002-initial-api-specification.rst

```diff
@@ -0,0 +1,436 @@
+       'title': 'The Learner Credit Subsidy for my Enterprise',
+       'subsidy_type': 'learner_credit',
+       'unit': 'USD_CENTS',
+       'opportunity_id': 'some-opp-id',
+       'remaining_balance': 987650
+   }
+
+Permissions
+-----------
+
+enterprise_admin
+  Should return the requested subsidy only if the requesting user has implicit (JWT) or explicit (DB-defined)
+  ``enterprise_admin`` role assigned for the requested subsidy's enterprise.
+
+openedx_operator
+  If the requesting user is implicitly or explicitly granted this role, they have permissions to view **all**
+  subsidy records, and can therefore retrieve the requested subsidy.
+
+Disallowed actions/verbs
+========================
+None of ``POST``, ``PATCH``, ``PUT``, or ``DELETE`` should be supported at this time.
+Currently, subsidy records must be created by a staff user via Django Admin. We don't support the creation
+of subsidies via a user-facing UI or from any other service.
+Neither deletions or modifications to a subsidy (e.g. the `title` field) are supported via a user-facing UI at this time.
+Any modifications that need to occur must be done by staff via Django Admin. Deletions of subsidies
+should not be enabled via the API (though perhaps are possible via Django Admin).
+
+Subsidy Transactions
+********************
+**/api/v1/subsidies/[subsidy-uuid]/transactions/**
+The root URL for reading aggregate and transaction-level data about the transactions
+associated with a subsidy.
+
+GET (list) enterprise subsidy transactions
+==========================================
+**/api/v1/subsidies/[subsidy-uuid]/transactions/?include_aggregates={true,false}**
+
+This endpoint can answer questions like:
+
+- Tell me all of the transactions for a given subsidy.
+- Within a subsidy, give me all transactions for a specific user or content key.
+- Given a user id and content key, tell me if a successful/committed transaction exists, that is,
+  has the user previously redeemed this subsidy for this content?
+
+Inputs
+------
+
+- ``subsidy-uuid`` (URL path, required): The uuid (primary key) of the subsidy for which transactions should be listed.
+- ``include_aggregates`` (query param, optional): Specifies if aggregates (quantities, number of transactions) should be
+  returned as part of the paginated response.  Defaults to ``true``.
+- ``user_id`` (query param, optional): If present, filters returned transactions and/or aggregates to only
+  those associated with the specified ``user_id`` value.
+- ``content_key`` (query param, optional): If present, filters returned transactions and/or aggregates to only
+  those associated with the specified ``content_key`` value.
+
+Outputs
+-------
+Returns a paginated list of aggregate and transaction data:
+
+::
+
+   {
+       'previous': null,
+       'next': '...',
+       'count': 'blah',
+       'aggregates': {
+           'total_quantity': 12350,
+           'unit': 'USD_CENTS',
```

Comment:
What if the resulting transactions contain a mix of units?  Should we somehow design for that eventuality, or just commit to only aggregating USD transactions for now?

Reply:
For now we should just commit to USD. What I think is likely to happen is that all transactions will be stored in a singular unit type and then when the combination of metadata about the transaction, contract data and historical currency exchange rates we would reconstruct the ledger in a different unit and then do the aggregation.

Reply:
I just noticed there's both a "total_quantity" and also a "remaining_balance".  Maybe that already answers my question, in that the former refers to seats and the latter refers to USD cents?